acct/chsh

```diff
@@ -1,38 +1,60 @@
-#!/usr/bin/env python
-from __future__ import print_function
-
+#!/usr/bin/env python3
+import argparse
 import getpass
-import os
+import subprocess
 import sys
 
-import ldap
-import ldap.modlist as modlist
-import ldap.sasl as sasl
-
-if len(sys.argv) < 2:
-    print('Usage: chsh shell')
-    sys.exit()
-
-shell = sys.argv[1]
-etcshells = open('/etc/shells', 'r')
-line = etcshells.readline()
-while line != '':
-    validshell = line.strip()
-    if shell == validshell:
-        print('Changing shell to %s' % (validshell))
-        kinit = '/usr/bin/klist -t || /usr/bin/kinit'
-        os.system(kinit)
-        conn = ldap.initialize('ldaps://ldap.ocf.berkeley.edu:636/')
-        auth = sasl.gssapi('')
-        conn.sasl_interactive_bind_s('', auth)
-        dn = 'uid=%s,ou=people,dc=ocf,dc=berkeley,dc=edu' % (getpass.getuser())
-        old = {'loginShell': '*'}
-        new = {'loginShell': '%s' % (validshell)}
-        ldif = modlist.modifyModlist(old, new)
-        conn.modify_s(dn, ldif)
-        conn.unbind_s()
-        print('Shell successfully changed, typically it takes a while for the '
-              'changes to propagate to all our machines')
-        sys.exit()
-    line = etcshells.readline()
-print('The shell you chose is not valid, please chose another shell')
+from ocflib.account.manage import modify_ldap_attributes
+from ocflib.misc.validators import VALID_LOGIN_SHELLS
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Change OCF login shell')
+    parser.add_argument(
+        '-s', '--shell',
+        help='new login shell for the user account',
+    )
+    args = parser.parse_args()
+
+    username = getpass.getuser()
+    print('Changing the login shell for {}'.format(username))
+
+    # If no shell was provided, enter interactive mode and prompt for a shell
+    if args.shell:
+        new_shell = args.shell
+    else:
+        print('Enter the new value, or press ENTER to keep your current shell')
```

Comment:
I would go with "Enter the new shell"

Reply:
I went with this since it's literally what the actual `chsh` prints, but I agree that `Enter the new shell` would be a clearer/better message:

```
$ chsh
Password: 
Changing the login shell for jvperrin
Enter the new value, or press ENTER for the default
	Login Shell [/usr/bin/zsh]:
```